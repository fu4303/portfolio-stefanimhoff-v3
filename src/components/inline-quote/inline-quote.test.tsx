import React from 'react';
import { axe } from 'jest-axe';

import { render } from '../../services/test-utils';

import { InlineQuote } from '.';
import type { InlineQuoteProps } from '.';

type InlineQuoteTestProps = InlineQuoteProps & {
  description: string;
};

describe('InlineQuote', () => {
  const combinations: InlineQuoteTestProps[] = [
    { description: 'default', children: 'Quote' },
    { description: 'lang en', children: 'Quote', lang: 'en' },
    { description: 'lang de', children: 'Quote', lang: 'de' },
    { description: 'lang ja', children: 'Quote', lang: 'ja' },
  ];

  combinations.forEach((combination) => {
    test(`renders in ${combination.description}`, () => {
      const { container } = render(<InlineQuote {...combination} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  combinations.forEach((combination) => {
    test(`is accessible in ${combination.description}`, async () => {
      const { container } = render(<InlineQuote {...combination} />);
      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });
  });
});
