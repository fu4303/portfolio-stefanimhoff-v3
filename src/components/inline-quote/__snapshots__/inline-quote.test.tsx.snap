// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineQuote renders in default 1`] = `
.c0 {
  font-style: italic;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0::before {
  font-style: normal;
  margin-right: 0.1em;
}

.c0::after {
  font-style: normal;
  margin-left: 0.1em;
}

<q
  class="c0"
  lang="en"
>
  Quote
</q>
`;

exports[`InlineQuote renders in lang de 1`] = `
.c0 {
  font-style: italic;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0::before {
  font-style: normal;
  margin-right: 0.1em;
}

.c0::after {
  font-style: normal;
  margin-left: 0.1em;
}

<q
  class="c0"
  lang="de"
>
  Quote
</q>
`;

exports[`InlineQuote renders in lang en 1`] = `
.c0 {
  font-style: italic;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0::before {
  font-style: normal;
  margin-right: 0.1em;
}

.c0::after {
  font-style: normal;
  margin-left: 0.1em;
}

<q
  class="c0"
  lang="en"
>
  Quote
</q>
`;

exports[`InlineQuote renders in lang ja 1`] = `
.c0 {
  font-style: italic;
}

.c0[lang|='ja'] {
  font-style: normal;
}

.c0::before {
  font-style: normal;
  margin-right: 0.1em;
}

.c0::after {
  font-style: normal;
  margin-left: 0.1em;
}

<q
  class="c0"
  lang="ja"
>
  Quote
</q>
`;
